### PR TITLE
Implement the Pigweed-app example on EFR32

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem
Need of a Pigweed rcp example on EFR32 platforms 

 #### Summary of Changes
- Create the example at `example/pigweed-app/efr32/`
- Init uart driver and create function to read and write a character
- link those  functions with pigweed backend `pw_sys_io`
- create GN build structure for pigweed usage with EFR32 platform
- move some files that can be shared for all EFR32 examples to efr32/platform
- Readme on how to build and run the pigweed-app

### CODE REVIEW

File: config/efr32/BUILD.gn
Comment (andy31415) [inline config/efr32/BUILD.gn:1]: 2021 here and for all new files

File: examples/persistent-storage/efr32/third_party/connectedhomeip
Comment (andy31415) [inline examples/persistent-storage/efr32/third_party/connectedhomeip:1]: Is this needed?
Path seems to be MacOS (?) specific. If needed, could we use a relative link instead?

File: examples/pigweed-app/efr32/BUILD.gn
Comment (andy31415) [inline examples/pigweed-app/efr32/BUILD.gn:30]: can the empty declare args be removed?

File: examples/persistent-storage/efr32/third_party/connectedhomeip
Comment (jmartinez-silabs) [reply examples/persistent-storage/efr32/third_party/connectedhomeip]: fixed to relative symlinks

File: examples/pigweed-app/efr32/BUILD.gn
Comment (jmartinez-silabs) [reply examples/pigweed-app/efr32/BUILD.gn]: removed

File: config/efr32/BUILD.gn
Comment (jmartinez-silabs) [reply config/efr32/BUILD.gn]: changed all new files to 2021

**Review by szatmz — APPROVED** [review]
Looks good. Please file an issue to move the SDK/platform logging from RTT over to pw_logs whenever pw_rpc is enabled. THank you!

File: examples/pigweed-app/efr32/BUILD.gn
Comment (jmartinez-silabs) [reply examples/pigweed-app/efr32/BUILD.gn]: done

**Review by andy31415 — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

File: config/efr32/BUILD.gn
Comment (mspang) [inline config/efr32/BUILD.gn:1]: What's the purpose of adding this file?

File: config/efr32/lib/pw_rpc/BUILD.gn
Comment (mspang) [inline config/efr32/lib/pw_rpc/BUILD.gn:37]: Why complete_static_lib ?

File: examples/lighting-app/efr32/args.gni
Comment (mspang) [inline examples/lighting-app/efr32/args.gni:26]: This won't work, it will always be false because args.gni is evaluated before the args are decided (it sets the defaults, cannot use args before they are set). I think you'll need to have multiple files

args.gni
args_pw_rpc.gni

and import one or the other into the command line.

**Review by mspang — CHANGES_REQUESTED** [review]

File: examples/lighting-app/efr32/args.gni
Comment (jmartinez-silabs) [reply examples/lighting-app/efr32/args.gni:26]: So when is the value of arguments from the command line actually applied to the declare_args() ?

Is there not a way to set build configuration base on argument variable?

File: config/efr32/lib/pw_rpc/BUILD.gn
Comment (jmartinez-silabs) [reply config/efr32/lib/pw_rpc/BUILD.gn:37]: No use, dragged this from a copy of a implemented static_library("pw_rpc") . I will delete it.

File: examples/lighting-app/efr32/args.gni
Comment (mspang) [reply examples/lighting-app/efr32/args.gni:26]: It's applied to declare_args() in all normal build files right when declare_args() is evaluated, the problem is that args.gni is _not_ a normal build file. It's a special build file contains defaults for arguments.

The default arguments is a single scope and cannot have conditions embedded in it. So we have to do something else, like have multiple args.gni and configure variants like this

`gn gen --args='import("//args/with_pw_rpc.gni")'`

```
// with_pw_rpc.gni
pw_log_BACKEND = "$dir_pw_log_basic"
pw_assert_BACKEND = "$dir_pw_assert_log"
```

**Review by woody-apple — APPROVED** [review]
Approving pending  @mspang 's feedback

File: config/efr32/BUILD.gn
Comment (jmartinez-silabs) [reply config/efr32/BUILD.gn]: remove, ended up not using it

File: examples/lighting-app/efr32/args.gni
Comment (jmartinez-silabs) [reply examples/lighting-app/efr32/args.gni]: I did the modification and added a with_pw_rpc.gni to the three EFR32 example app. Those can be imported in args when we will want to build one of these examples with pw_rpc. 

pw_rpc for lock and light examples is still a WiP. I need to bridge standard logs to the pw_logs. This is my next step once the pigweed-app example is merged.

**Review by mspang — APPROVED** [review]

**Comment by jmartinez-silabs** [comment]
@woody-apple @andy31415 @mspang Can this be Merged?

**Pull request merged**
